torch/nn/modules/module.py

```diff
@@ -1312,7 +1314,13 @@ def _load_from_state_dict(self, state_dict, prefix, local_metadata, strict,
                 :meth:`~torch.nn.Module.load_state_dict`
         """
         for hook in self._load_state_dict_pre_hooks.values():
-            hook(state_dict, prefix, local_metadata, strict, missing_keys, unexpected_keys, error_msgs)
+            n_params = len(signature(hook).parameters)
+            if n_params == 8:
+                # Pass in module instance as well.
+                hook(state_dict, prefix, local_metadata, strict, missing_keys, unexpected_keys, error_msgs, self)
```

Comment:
This is a bit hacky, another option I had in mind was to have an additional option in _register_load_state_dict_pre_hook something like `with_module` and store that option with the hook. Then during load time, based on that option we either pass in the module or we don't for the hook.